### ISSUE TITLE
Successfully distributed build to Internal testers, but build number in testflight is: Ready to submit

### ISSUE THREAD

**Issue by bachvanthe1994** [fastlane/fastlane#17744]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [x ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x ] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x ] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Successfully distributed build to Internal testers, but build number in the Testflight is: Ready to submit

![Screenshot 2020-12-08 at 15 11 05](https://user-images.githubusercontent.com/11937975/101457988-d18d1c00-3968-11eb-9694-877e313bb2bb.png)
![Screenshot 2020-12-08 at 15 12 52](https://user-images.githubusercontent.com/11937975/101457998-d3ef7600-3968-11eb-98ef-82aabf26bec0.png)
![image](https://user-images.githubusercontent.com/11937975/101458188-2335a680-3969-11eb-8d4c-800c6a6b5bc3.png)
![image](https://user-images.githubusercontent.com/11937975/101458220-2d57a500-3969-11eb-8330-006ec4e01fea.png)

##### Command executed
[12:57:31]: [36m$ bundle exec fastlane lane_upload_to_testflight workspaceFolderPath:/Users/iosteam1/Jenkins/workspace/IVB/IVB Testflight Test/ ipaIPAFilePath:/Users/iosteam1/Jenkins/workspace/IVB/IVB Testflight Test/fastlane_build/ProjectA.ipa ipaBundleIdentifier:******* ipaTeamIDITC:******* testflightUserName:5f2bac3d18579ae50c9168e613c9369a testflightPass:e1fb673e4f46249b71a70cc42a044df2299838883f5745e97ac6722db11ecc7e testflightSpecificPass:ac7d2dc576c31bdf759a347d952111eea3d0b47080ce90923c4bab4f97d3d1a4 testflightContactEmail:******** testflightContactFirstName:Nguyễn testflightContactLastName:Tiến testflightContactPhone:********* testflightGroups:WyJJVkIgVGVzdCBHcm91cCJd testflightWaitProcessingInterval:300 testflightChangelog:IOS bàn giao lãi suất cơ sở [0m
[12:57:31]: 
[12:57:31]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
[12:57:33]: [32mDriving the lane 'lane_upload_to_testflight' 🚀[0m
[12:57:33]: [32m----------------------------------[0m
[12:57:33]: [32m--- Step: upload_to_testflight ---[0m
[12:57:33]: [32m----------------------------------[0m
[12:57:33]: Login to App Store Connect (*********)
[12:57:35]: Login successful
[12:57:38]: [32mReady to upload new build to TestFlight (App: 1096963960)...[0m
[12:57:38]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
[12:57:51]: [32miTunes Transporter successfully finished its job[0m
[12:57:51]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
[12:57:51]: Going to upload updated app to App Store Connect
[12:57:51]: [32mThis might take a few minutes. Please don't interrupt the script.[0m
[12:58:35]: [32miTunes Transporter successfully finished its job[0m
[12:58:35]: [32m--------------------------------------------------------------------[0m
[12:58:35]: [32mSuccessfully uploaded package to App Store Connect. It might take a few minutes until it's visible online.[0m
[12:58:35]: [32m--------------------------------------------------------------------[0m
[12:58:35]: [32mSuccessfully uploaded the new binary to App Store Connect[0m
[12:58:35]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[12:58:35]: Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps.
[12:58:35]: Waiting for processing on... app_id: 1096963960, app_version: 9.9.9, build_version: 129, platform: IOS
[12:58:36]: [33mRead more information on why this build isn't showing up yet - https://github.com/fastlane/fastlane/issues/14997[0m
[12:58:36]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[13:03:39]: Waiting for App Store Connect to finish processing the new build (9.9.9 - 129) for IOS
[13:08:40]: [32mSuccessfully finished processing the build 9.9.9 - 129 for IOS[0m
[13:08:42]: [32mSuccessfully set the changelog for build[0m
[13:08:42]: Distributing new build to testers: 9.9.9 - 129
[13:08:43]: [33mExport compliance has been set to 'false'. Need to wait for build to finishing processing again...[0m
[13:08:43]: [33mSet 'ITSAppUsesNonExemptEncryption' in the 'Info.plist' to skip this step and speed up the submission[0m
[13:08:43]: Waiting for processing on... app_id: 1096963960, app_version: 9.9.9, build_version: 129, platform: IOS
[13:08:44]: [32mSuccessfully finished processing the build 9.9.9 - 129 for IOS[0m
[13:08:44]: Build 9.9.9 - 129 already submitted for review
[13:08:45]: [32mSuccessfully distributed build to Internal testers 🚀[0m

+------+----------------------+-------------+
|             [32mfastlane summary[0m              |
+------+----------------------+-------------+
| Step | Action               | Time (in s) |
+------+----------------------+-------------+
| 1    | upload_to_testflight | 671         |
+------+----------------------+-------------+

[13:08:45]: [32mfastlane.tools just saved you 11 minutes! 🎉[0m

#######################################################################
# fastlane 2.170.0 is available. You are on 2.169.0.
# You should use the latest version.
# Please update using `sudo gem install fastlane`.
#######################################################################
# Run `sudo gem cleanup` from time to time to speed up fastlane
#######################################################################

[32m2.170.0 Improvements and Upload App Privacy Details[0m
* [Fastlane.swift] fixed potential undefined method `action_name' in generator (#17660) via phil zhang
* [action] onesignal - update onesignal.rb to include optional organization_id (#17696) via Brandon Carpenter
* [deliver] fix a regression in language detection in Deliver::UploadMetadata (#17730) via Satoshi Namai
* [action] NEW ACTION for App Data Usages: upload_app_privacy_details_to_app_store (#17592) via Josh Holtz
* [match] only include enabled devices for force_for_new_devices (#17695) via Steven Roebert

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by max-ott** [comment]
Please include all information required in the issue template and re-open your issue.

**Comment by bachvanthe1994** [comment]
> Please include all information required in the issue template and re-open your issue.

I added information for you, please check for me.

**Comment by bachvanthe1994** [comment]
At the moment, my company using Fastlane for more than 20 projects iOS, I need to resolve this issue soon as soon, if you want more information please tell me, I will support you.

**Comment by max-ott** [comment]
This is still an open source project. If you need immediate support look for a `fastlane` freelancer and pay him/her.

**Comment by max-ott** [comment]
Btw just checking the logs: This means the build is distributed to internal testers (if they got added). The "ready to submit" means, you can submit it for TestFlight review (external). Please clarify, what you are actually expecting.

**Comment by bachvanthe1994** [comment]
> Btw just checking the logs: This means the build is distributed to internal testers (if they got added). The "ready to submit" means, you can submit it for TestFlight review (external). Please clarify, what you are actually expecting.

This case happens sometimes. Sometimes this build version will release and notify Tester success, but sometimes it errors, stop In case "ready to submit", if normal, I will see status "Testing" in the Testflight.
This case happened recently.

**Comment by bachvanthe1994** [comment]
But func upload_to_testflight always return: Successfully distributed build to Internal testers 🚀
:(

**Comment by bachvanthe1994** [comment]
![image](https://user-images.githubusercontent.com/11937975/102601924-6525d000-4153-11eb-8d24-bbe49faab735.png)

to fix this bug
we need call this func 
client.post_beta_app_review_submissions(build_id: id)

after call: uploaded_build.add_beta_groups(beta_groups: beta_groups)

i hope in the last version fastlane will append this func

**Comment by rogerluan** [comment]
@bachvanthe1994 although this would resolve your issue:

<img width="843" alt="image" src="https://user-images.githubusercontent.com/8419048/102688587-73690000-41d6-11eb-92ac-089a68c28c8b.png">


Sounds like you want to submit your app to external testers (that's what `post_beta_app_review_submission` does). And if that's the case, the issue can be fixed by passing `distribute_external: true` to `pilot`. Can you try that?

**Comment by bachvanthe1994** [comment]
> @bachvanthe1994 although this would resolve your issue:
> 
> <img alt="image" width="843" src="https://user-images.githubusercontent.com/8419048/102688587-73690000-41d6-11eb-92ac-089a68c28c8b.png">
> 
> Sounds like you want to submit your app to external testers (that's what `post_beta_app_review_submission` does). And if that's the case, the issue can be fixed by passing `distribute_external: true` to `pilot`. Can you try that?
> 
> https://docs.fastlane.tools/actions/pilot/

Thanks bro, in the any case, we need call post beta review submission after done all step, after that, the builder will display status Testing

**Comment by rogerluan** [comment]
@bachvanthe1994 right, so does setting `distribute_external: true` work for you?

**Comment by bachvanthe1994** [comment]
> distribute_external: true

No bro.

**Comment by bachvanthe1994** [comment]
> @bachvanthe1994 although this would resolve your issue:
> 
> <img alt="image" width="843" src="https://user-images.githubusercontent.com/8419048/102688587-73690000-41d6-11eb-92ac-089a68c28c8b.png">
> 
> Sounds like you want to submit your app to external testers (that's what `post_beta_app_review_submission` does). And if that's the case, the issue can be fixed by passing `distribute_external: true` to `pilot`. Can you try that?
> 
> https://docs.fastlane.tools/actions/pilot/

this is the same code.

**Comment by iostaf** [comment]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I have almost the same issue. When I do automatic distribution to the external testers group (named 'Internal Testing') it appears in the App Store Connect with 'Ready to Submit' instead of 'Testing' status. The 'Internal Testing' group is added to the build.
As a workaround I need to manually remove the 'Internal Testing' group and readd it again. After it moves to the Testing status.
Note: It used to work before. Not sure from which version of fastlane it started not to work but I've experienced this issue for 2 weeks at least.

##### Command executed
``` shell
fastlane pilot upload -u *** \
-i ./_files/MyIPA.ipa \
--changelog 'Bugfixes and stability improvements' \
--distribute_external true \
--groups 'Internal Testing'
```

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>
[✔] 🚀
[12:45:19]: fastlane detected a Gemfile in the current directory
[12:45:19]: However, it seems like you didn't use `bundle exec`
[12:45:19]: To launch fastlane faster, please use
[12:45:19]:
[12:45:19]: $ bundle exec fastlane pilot upload -u *** -i ./_files/MyIPA.ipa --changelog Bugfixes and stability improvements --distribute_external true --groups Internal Testing
[12:45:19]:
[12:45:19]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[12:45:22]: Login to App Store Connect (***)
[12:45:24]: Login successful
[12:45:28]: Ready to upload new build to TestFlight (App: ***)...
[12:45:28]: Going to upload updated app to App Store Connect
[12:45:28]: This might take a few minutes. Please don't interrupt the script.
[12:46:19]: [Transporter Warning Output]: WARNING ITMS-90863: "Apple silicon Macs support issue. The app contains the following UIRequiredDeviceCapabilities values. These capabilities are not available on Mac. [telephony]"
[12:46:19]: iTunes Transporter successfully finished its job
[12:46:19]: WARNING ITMS-90863: "Apple silicon Macs support issue. The app contains the following UIRequiredDeviceCapabilities values. These capabilities are not available on Mac. [telephony]"
[12:46:19]: ------------------------------------------------------------------------------------------------------------------
[12:46:19]: --- Successfully uploaded package to App Store Connect. It might take a few minutes until it's visible online. ---
[12:46:19]: ------------------------------------------------------------------------------------------------------------------
[12:46:19]: Successfully uploaded the new binary to App Store Connect
[12:46:19]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[12:46:19]: Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps.
[12:46:19]: Waiting for processing on... app_id: ***, app_version: ***, build_version: 42517, platform: IOS
[12:46:20]: Read more information on why this build isn't showing up yet - https://github.com/fastlane/fastlane/issues/14997
[12:46:20]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:46:50]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:47:21]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:47:51]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:48:22]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:48:52]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:49:23]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:49:53]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:50:24]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:50:54]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[12:51:28]: Waiting for App Store Connect to finish processing the new build (*** - 42517) for IOS
[12:57:06]: Waiting for App Store Connect to finish processing the new build (*** - 42517) for IOS
[12:57:37]: Successfully finished processing the build *** - 42517 for IOS
[12:57:38]: Successfully set the changelog for build
[12:57:39]: Distributing new build to testers: *** - 42517
[12:57:39]: Export compliance has been set to 'false'. Need to wait for build to finishing processing again...
[12:57:39]: Set 'ITSAppUsesNonExemptEncryption' in the 'Info.plist' to skip this step and speed up the submission
[12:57:39]: Waiting for processing on... app_id: ***, app_version: ***, build_version: 42517, platform: IOS
[12:57:40]: Successfully finished processing the build *** - 42517 for IOS
[12:57:40]: Build *** - 42517 already submitted for review
[12:57:41]: Successfully distributed build to External testers 🚀  
  </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 10.15.7                                          |
| Ruby                        | 2.6.0                                            |
| Bundler?                    | false                                            |
| Git                         | git version 2.29.1                               |
| Installation Source         | ~/.rbenv/versions/2.6.0/bin/fastlane             |
| Host                        | Mac OS X 10.15.7 (19H114)                        |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.0/lib                      |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode_12.3.app/Contents/Developer/ |
| Xcode Version               | 12.3                                             |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.172.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.2        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.16.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| artifactory               | 3.0.15       |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.73.0       |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 1.0.1        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 1.17.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.14.1       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.0        |
| googleauth                | 0.12.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-core         | 1.5.0        |
| digest-crc                | 0.5.1        |
| google-cloud-storage      | 1.26.0       |
| emoji_regex               | 1.0.1        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.296.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.1.1        |
| aws-sdk-core              | 3.94.0       |
| aws-sdk-kms               | 1.30.0       |
| aws-sdk-s3                | 1.61.2       |
| json                      | 2.3.0        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| date                      | 1.0.0        |
| fileutils                 | 1.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2021-01-26**

**Comment by sachajln** [comment]
I can confirm we have the same issue on my side, our apps are distributed with `distribute_external: true` but are sometime randomly put in a `Ready to submit` state. It started in late november / early december.

**Comment by victorcatao** [comment]
My team has the same problem. We are setting `distribute_external: true`, but it does not appear the status "Testing", despite distributing it to external users. We need to manually remove the external tester's group and add it again.
This started happening recently, without making any changes to the fastlane settings :(

**Comment by ricardohochman** [comment]
Hi guys! Any updates here? I'm also facing the same issue

**Comment by eduhcano** [comment]
Hi, I just face the same problem, distribute_external: true but build is _Ready to Submit._

**Comment by albertreed** [comment]
Has anyone found a workaround?

**Comment by bachvanthe1994** [comment]
> ![image](https://user-images.githubusercontent.com/11937975/102601924-6525d000-4153-11eb-8d24-bbe49faab735.png)
> 
> to fix this bug
> we need call this func
> client.post_beta_app_review_submissions(build_id: id)
> 
> after call: uploaded_build.add_beta_groups(beta_groups: beta_groups)
> 
> i hope in the last version fastlane will append this func
> 
> file: build.rb
> 
> ![image](https://user-images.githubusercontent.com/11937975/102602135-afa74c80-4153-11eb-83d0-d2d692d4a649.png)

@albertreed you can fix this issue with my guide above.

**Comment by joshdholtz** [comment]
Hey, fam! I got a fix for this over at https://github.com/fastlane/fastlane/pull/18328

It turns out this issue was due to setting the export compliance settings 😬  When setting export compliance settings via the App Store Connect API, App Store Connect needs to reprocess the builds. `pilot` was sometimes returning the processed build too soon which put things into a weird state when trying to submit for review and adding groups. This PR properly waits and returns the build when its actually ready to be submitted 🤷‍♂️
